### PR TITLE
feat: additional vec apis

### PR DESCRIPTION
**Pull request by amunra**
# Overview
Improved `Vec` implementation.

## New Associated Functions
* `allocator`
* `truncate`
* `extend_with`
* `resize`
* `resize_with`

## New Traits
* `PartialEq`
* `AsRef`
* `AsMut`

Closes https://github.com/questdb/alloc-checked/issues/6

### CODE REVIEW

File: src/vec.rs [hunk 1]
```diff
@@ -17,6 +18,17 @@ impl<T, A: Allocator> Vec<T, A> {
         }
     }
 
+    pub fn allocator(&self) -> &A {
+        self.inner.allocator()
+    }
+
+    #[inline]
+    pub fn try_with_capacity_in(capacity: usize, alloc: A) -> Result<Self, TryReserveError> {
```
Comment (eugenels) [inline src/vec.rs:26]: what is the diff with `with_capacity_in` ?

File: src/vec.rs [hunk 2]
```diff
@@ -485,4 +612,228 @@ mod tests {
         }
         assert_eq!(wma.in_use(), 0);
     }
+
+    #[test]
+    fn try_with_capacity_in() {
```
Comment (eugenels) [inline src/vec.rs:617]: It looks like most of the tests stay within the memory limit. I'd suggest adding an over-limit case to make sure the TryReserveError isn’t silently ignored in the implementation.

File: src/vec.rs (hunk 1 above)
Comment (amunra) [reply src/vec.rs]: Oops!

I didn't realise I had that method implemented already.
`with_capacity_in` is outright a better name - more in line with the rest of the naming.

Deleting the redundant method, but keeping the new impl.

File: src/vec.rs (hunk 2 above)
Comment (amunra) [reply src/vec.rs]: Deleted this test, augmented the test for `with_capacity_in` which already tested that branch.

**Review by eugenels — APPROVED** [review]

**Pull request merged**